eli5/xgboost.py

```diff
@@ -217,6 +214,136 @@ def explain_prediction_xgboost(
      )
 
 
+@explain_prediction.register(XGBClassifier)
+@explain_prediction.register(XGBRegressor)
+@explain_prediction.register(Booster)
+def explain_shap_prediction_xgboost(
```

Comment:
`explain_prediction_xgboost` and `explain_shap_prediction_xgboost` shares a lot of lines of code. Avoid code duplication for example by creating a function that accepts as parameter a function to use for retrieving the weights.

Reply:
@ivanprado I did not make a function, since I thought that previous `explain_prediction` will be deprecated. Now if we intend to keep it, I have a suggestion. So by default we can have shap prediction but maybe adding an extra parameter to let the user choose between the type of explanation. What do you think? @ivanprado

Reply:
Sounds good to me.

Reply:
Besides test should be then updated to run them for both versions just by adding a new `pytest.mark.parametrize`

Reply:
@ivanprado  Yes then I will include a parameter and include parametrize